test/integration/mapping_validation_test.rb

```diff
@@ -20,6 +20,35 @@ class MappingValidationTest < ActiveSupport::TestCase
       #{invalid_categories}"
   end
 
+  test "every Shopify category has corresponding channel mappings" do
+    mappings_json_data = CLI.new.parse_json("dist/en/integrations/all_mappings.json")
+    shopify_category_lack_mappings = []
+
+    mappings_json_data["mappings"].each do |mapping|
+      next unless mapping["input_taxonomy"].include?("shopify")
+
+      all_shopify_category_ids = category_ids_from_taxonomy(mapping["input_taxonomy"])
+      next if all_shopify_category_ids.nil?
+
+      shopify_category_ids_from_mappings_input = Set.new
+      mapping["rules"].each do |rule|
+        shopify_category_ids_from_mappings_input.add(rule["input"]["category"]["id"])
+      end
+      shopify_category_ids_diff = all_shopify_category_ids - shopify_category_ids_from_mappings_input
+
+      shopify_category_lack_mappings << {
+        input_taxonomy: mapping["input_taxonomy"],
+        output_taxonomy: mapping["output_taxonomy"],
+        count_of_shopify_category_ids_lack_mappings: shopify_category_ids_diff.size,
+        shopify_category_ids_lack_mappings: shopify_category_ids_diff,
+      }
+    end
+
+    assert shopify_category_lack_mappings.empty?,
+      "The following shopify category ids lack corresponding channel mappings:
+    #{shopify_category_lack_mappings}"
```

Comment:
```suggestion
    assert_empty(
      shopify_category_lack_mappings,
      "The following shopify category ids lack corresponding channel mappings: #{shopify_category_lack_mappings}"
    )
```

Reply:
I also wonder if we should do this _in the loop_ vs collecting all of them? Or maybe fail on first to save time when there is failure? 🤔

Reply:
Users need to get a full list of missing mappings to correct the issue throughly and they can run this test locally to avoid unnecessary CI checks.